crates/bindings-csharp/Codegen/Module.cs

```diff
@@ -31,18 +33,29 @@ bool isEquatable
     )
         : base(name, type, typeInfo)
     {
-        Attrs = attrs;
+        (string?, ColumnAttrs)[] x = [(default(string), attrs)];
+        Attrs = new(x.ToImmutableArray());
         IsEquatable = isEquatable;
+        FullTableName = tableName;
     }
 
-    public ColumnDeclaration(IFieldSymbol field)
+    public ColumnDeclaration(string tableName, IFieldSymbol field)
         : base(field)
     {
-        Attrs = field
+        FullTableName = tableName;
+
+        Attrs = new(field
             .GetAttributes()
-            .Where(a => a.AttributeClass?.ToString() == "SpacetimeDB.ColumnAttribute")
-            .Select(a => (ColumnAttrs)a.ConstructorArguments[0].Value!)
-            .SingleOrDefault();
+            .Select(a => (a.NamedArguments.FirstOrDefault(a => a.Key == "Table").Value.Value as string,
+                a.AttributeClass?.ToString() switch
+                {
+                    "SpacetimeDB.AutoIncAttribute" => ColumnAttrs.AutoInc,
+                    "SpacetimeDB.PrimaryKeyAttribute" => ColumnAttrs.PrimaryKey,
+                    "SpacetimeDB.UniqueAttribute" => ColumnAttrs.Unique,
+                    "SpacetimeDB.IndexedAttribute" => ColumnAttrs.Indexed,
+                    _ => ColumnAttrs.UnSet,
```

Comment:
Looks like this will even store separate array entries for 3rd-party attributes that don't have anything to do with SpacetimeDB. Those entries will be pretty much useless `(null, UnSet)`, so I'd suggest skipping them over to improve cacheability.